packages/components/aside/src/aside.js

```diff
@@ -3,49 +3,37 @@ import css from './aside.css';
 import '@tradeshift/elements.button';
 import '@tradeshift/elements.cover';
 import '@tradeshift/elements.spinner';
-import { customEventNames } from './utils';
```

Comment:
I personally would like to have these event names as an exportable enum, because IMHO it improves the dev experience and makes a code less error-prone, when you use enum values instead of plain string values. And it brings additional value for typescript projects - with enum they will have a compile-time check that the event name actually exists.
Example of how I think it can be used in a separate project:
https://github.com/aleks-elkin/react-typescript-web-components/blob/master/src/components/ElementsModal.tsx#L24

Also, with separate enum we could add a JSDoc comment on top of each item, which could be useful for documentation as well.

Reply:
I also like them and think this pattern can be useful but after introducing auto-generated prop from resource, I believe they can be actually generated to help other developers for using the event name, slot name and etc. I think removing them is helping with keeping everything in one place and make it simpler for people to contribute and maintain. These were (and some still are) inconsistent part of the library that can be fixed by removing unnecessary convention, so we don't miss it in review process.

I think we should keep all the related information about a slot or event in one place and document it in the same place, but generate those files for developers using the library, to don't loose the benefits of them. This should be handled in the same PR that we'll generate the types files. What do you think?

Reply:
I'm a bit afraid that source code could became a mess when we have documentation comments in the middle of the implementation, but it is just a guess. Let's try to do it in your way and see how it will look.

Another corner case to think about: what to do when we dispatch the same event in several places? Should we document all of them or just the first one?
- If only one would be enough, but then a developer have to keep a track when each event is introduced for the first time.
- if all should be documented, we potentially could have an inconsistency because people make mistakes and could forget to update all docs.

Reply:
I agree that it can get really messy and personally don't like code comments in general, no matter what they've being used for, but there's a trade-off that I think at least for now it worth the try. Let's get back to this after updating all descriptions and other info in those comments.
Great point! 🤔 
I also believe it would rare cases that happen in the library and especially in smaller/simpler components that won't happen. My suggestion for those cases is a disable next line comment (eslint, ... ). What do you say? I can add it to this PR.

Reply:
Ok, let's try your approach to put comments in implementation and see how it will work out :)

I don't understand your suggestion about eslint, can you elaborate a bit on it?
To clarify, my concern is what to do when we have such situation:

```js
class TSComponent extends TSElement {
  ...
   someFunc () {
    this.dispatchCustomEvent('some-event-name', {data: "someData"});
  }
  ...
   anotherFunc () {
    this.dispatchCustomEvent('some-event-name', {data: "someData"});
  }
}
```
We have two functions that dispatch the same event.
Should the developer put JSDoc comment about event only in first case or put it on both events?
And what to do if after refactoring we swap these functions and `anotherFunc` became higher than `someFunc`?

It is not a blocker for this PR, just a corner case that I could imagine.

Reply:
It should be only one of them I would say, but not important which one.
I think the script should get improved and cover these cases and if there are two instances with jsdoc, throw an error to let developers know that there's another place that this event being documented.